### DISCUSSION TITLE
Permanently visible footer

### DISCUSSION THREAD

**Discussion by pideu-mh** [squidfunk/mkdocs-material#5058] [Q&A] [answered]
As far as I can tell there is no configuration option to make the footer permanently visible.

I was able to create an approximation of a permanently visible footer like this:
```
.md-footer
{
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%
}
```

One problem is that now the last few lines of a longer page are not visible any more, they are hidden by the footer.

I could basically fix this by also adding:
```
.md-container
{
  margin-bottom: 5em;
}
```
This is kind of ugly, because it does not take into account the height of the footer at all. If that height is modified, this has to be changed as well.

While all this basically works, there seem to be some edge cases that do not behave correctly.

For example, on a page that has some long scrollable contents and a table of contents that is also very long and scrollable, scrolling to the very bottom of the page reveals some overlay problem - the very end of the table of contents is partially obscured at the very bottom of the page, but when you scroll up, it appears again. It's hard to describe, not very noticable, but still a problem.

So the question is: 

Is there another, maybe better solution to achieve a permanently visible footer? 

Maybe this could be implemented at a lower level correctly and then a configuration option can be added?

**Comment by squidfunk** [comment]
Why would you want a permanently visible footer?

**Reply by pideu-mh** [reply]
For work I am experimenting with mkdocs & mkdocs-material for some technical documentation that may be put on a public-facing website. The marketing people had two requests: add some links to the footer and make it smaller & permanently visible.

Who am I to judge? After I made it smaller by setting the padding of `.md-social` to 0 it does not occupy too much screen estate any more and looks ok to me.

**Reply by squidfunk** [reply]
I understand. Please note that it's nothing we consider adding to Material for MkDocs itself, as it's also the very first time somebody brought this idea up, ever. I've also never seen something like this on other websites, as it just eats screen estate.

**Comment by pideu-mh** [comment, marked as answer]
> Please note that it's nothing we consider adding to Material for MkDocs itself, as it's also the very first time somebody brought this idea up, ever.

Thanks for pointing this out, although I was kind of surprised that nobody asked this before.

Ok, so I conclude that I am on my own here and have to see if the solution I came up with is good enough.

And it is good that this idea is now documented here, so when somebody has the same idea in the future and searches for it, he can come here and post a better solution. 😃

**Reply by squidfunk** [reply]
Exactly, thanks for sharing your solution!

**Reply by ProsperousHeart** [reply]
You're not on your own :)
